src/nvim/generators/gen_api_ui_events.lua

```diff
@@ -50,6 +51,108 @@ function write_arglist(output, ev, need_copy)
   end
 end
 
+function extract_and_write_arglist(output, ev)
+  local hlattrs_args_count = 0
+  for j = 1, #ev.parameters do
+    local param = ev.parameters[j]
+    local kind = param[1]
+    output:write('  '..kind..' arg_'..j..' = ')
+    if kind == 'HlAttrs' then
+      -- The first HlAttrs argument is rgb_attrs and second is cterm_attrs
+      output:write('dict2hlattrs(args.items['..(j-1)..'].data.dictionary, '..(hlattrs_args_count == 0 and 'true' or 'false')..');\n')
+      hlattrs_args_count = hlattrs_args_count + 1
+    elseif kind == 'Object' then
+      output:write('args.items['..(j-1)..'];\n')
+    else
+      output:write('args.items['..(j-1)..'].data.'..string.lower(kind)..';\n')
+    end
+  end
+end
+
+function call_ui_event_method(output, ev)
+  output:write('  ui_call_'..ev.name..'(')
+  for j = 1, #ev.parameters do
+    output:write('arg_'..j)
+    if j ~= #ev.parameters then
+      output:write(', ')
+    end
+  end
+  output:write(');\n')
+end
+
+function make_raw_line_and_call(output)
+  -- arg_1 = grid, arg_2 = row, arg_3 = startcol, arg_4 = cells, arg_5 = endcol, arg_6 = clearcol, arg_7 = clearattr
+  
+  output:write([[
+  Integer arg_5, arg_6, arg_7;
+  LineFlags lineflags = kLineFlagWrap;
```

Comment:
We need to add the wrap flag to `grid_line`. But until then we should use `lineflags = 0`.

Reply:
Okay. And that solves the issue of last column being one less than that of server too. :)